tabrepo/benchmark/task/openml/task_utils.py

```diff
@@ -1,11 +1,120 @@
 from __future__ import annotations
 
-import time
 
+import boto3
+import logging
+import os
 import openml
 import pandas as pd
+import time
+import xmltodict
+from pathlib import Path
+
 from openml import OpenMLSupervisedTask
 from openml.exceptions import OpenMLServerException
+from tabflow.utils.s3_utils import parse_s3_uri
+
+logger = logging.getLogger(__name__)
+
+
+def download_task_from_s3(task_id: int, s3_dataset_cache: str = None) -> bool:
+    """
+    Downloads the task and dataset from S3 if available.
+    """
+    if s3_dataset_cache is None:
+        return False
+
+    # OpenML cache directory location for tasks + datasets
+    local_cache_dir = Path(os.path.expanduser("~/.cache/openml/org/openml/www"))
+    os.makedirs(local_cache_dir, exist_ok=True)
+    
+    try:
+        s3_client = boto3.client('s3')
+        s3_bucket, s3_prefix = parse_s3_uri(s3_uri=s3_dataset_cache)
+
+        task_cache_dir = local_cache_dir /  "tasks" / str(task_id)
+        os.makedirs(task_cache_dir, exist_ok=True)
+
+        logger.info(f"Attempting to download task {task_id} from S3 bucket {s3_bucket}")
+        s3_key_prefix = f"{s3_prefix}/tasks/{task_id}/org/openml/www/tasks/{task_id}"
+        
+        try:
+            s3_client.download_file(
+                Bucket=s3_bucket,
+                Key=f"{s3_key_prefix}/task.xml",
+                Filename=str(task_cache_dir / "task.xml")
+            )
+            logger.info(f"Downloaded task.xml for task {task_id} from S3")
```

Comment:
Don't download the files if they already exist locally.

Ditto for all other instances